scripts/install

```diff
@@ -13,82 +36,147 @@ set -eu
+
+# Calculate and store the config file checksum into the app settings
+#ynh_store_file_checksum "$final_path/conf/local.php"
+#=================================================
 
-# Get source
-SETUP_SOURCE
+#=================================================
+# SETUP SYSTEMD
+#=================================================
 
-sudo cp ../conf/dokuwiki.php $final_path/conf
-sudo cp ../conf/acl.auth.php $final_path/conf
+# Create a dedicated systemd config
+#ynh_add_systemd_config
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
 
 # Files owned by dokuwiki can just read
-sudo chown -R root: $final_path
+chown -R root: $final_path
 
 # except for conf, data, some data subfolders, and lib/plugin, where www-data must have write permissions
-sudo chown -R $app:root $final_path/{conf,data,data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp,lib/plugins,lib/tpl}
-sudo chmod -R 700 $final_path/conf
-sudo chmod -R 700 $final_path/data
-sudo chmod -R 755 $final_path/lib/plugins
-sudo chmod 755 $final_path/lib/tpl/{dokuwiki,dokuwiki/images}
-
-# Modify Nginx configuration file and copy it to Nginx conf directory
-sudo cp ../conf/nginx.conf /etc/nginx/conf.d/$domain.d/$app.conf
-sudo sed -i "s@__PATHTOCHANGE__@$path@g" /etc/nginx/conf.d/$domain.d/$app.conf
-sudo sed -i "s@__FINALPATH__@$final_path@g" /etc/nginx/conf.d/$domain.d/$app.conf
-sudo sed -i "s@__NAMETOCHANGE__@$app@g" /etc/nginx/conf.d/$domain.d/$app.conf
-
-if [ "$is_public" = "Yes" ];
-then
-	sudo sed -i "s@#--PRIVATE--@@g" /etc/nginx/conf.d/$domain.d/$app.conf
-fi
+chown -R $app:root $final_path/{conf,data,data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp,lib/plugins,lib/tpl}
+chmod -R 700 $final_path/conf
+chmod -R 700 $final_path/data
+chmod -R 755 $final_path/lib/plugins
+chmod 755 $final_path/lib/tpl/{dokuwiki,dokuwiki/images}
+
+#=================================================
+# SETUP LOGROTATE
+#=================================================
+
+# Use logrotate to manage application logfile(s)
+ynh_use_logrotate
 
-# Create the php-fpm pool config
-POOL_FPM
+#=================================================
+# ADVERTISE SERVICE IN ADMIN PANEL
+#=================================================
 
-# If app is public, add url to SSOWat conf as skipped_uris
-if [[ $is_public -eq 1 ]]; then
-  # unprotected_uris allows SSO credentials to be passed anyway.
-  ynh_app_setting_set "$app" unprotected_uris "/"
+#yunohost service add NAME_INIT.D --log "/var/log/FILE.log"
+yunohost service add "$app" --log "/var/log/$app/$app.log"
+
+#=================================================
+# SETUP SSOWAT
+#=================================================
+
+if [ $is_public -eq 0 ]
+then	# Remove the public access
+	ynh_app_setting_delete $app skipped_uris
```

Comment:
No need to do that, no skipped_uris have been added before.

Reply:
I'm not sure to understand. This line will removed the 'skipped_uris' entry in the settings.yml, but there isn't any, nowhere in the install script there's any `ynh_app_setting_set $app skipped_uris ...` so it's useless to try to remove it.

Reply:
I Now understand that it is technically not needed to do this test thanks to your explanation.

My comment is oriented for the next human / fresh packager that will read this code later. He might ask himself why this part is not in code while it is in the example package https://github.com/YunoHost/example_ynh/blob/master/scripts/install#L198:L201

It is not explained there. Maybe somewhere in the documentation on the Yunohost website?

Reply:
Yes, again I agree with you that this example app really needs a lot of explanations.
I'm going to try to find some time to do it, I understand that it's really importan.